### DISCUSSION TITLE
Looking for ESP32 in Arduino UNO form factor with integrated 5V shifter

### DISCUSSION THREAD

**Discussion by cesar9270** [awawa-dev/HyperSerialESP32#31] [Q&A]
Hello,

I am new to ESP32 controller.
I am using a classic arduino UNO ATmega328 connected through USB serial to my Windws PC / HyperHDR to drive my TV ambilight led strip. 
I would like to give a try on ESP32 chip.
I see the proposed ESP32 wire schema [here ](https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html) with a 5V level shitfer to drive the LED strip and DC 5V connector. Components are shown soldered on a generic prototyping board.

My question : is there any all-in-one board sold on aliexpress with an ESP32 in Arduino UNO form factor and 5V logic output so I can re-use my controller case and connectors (DC 5V , USB mini or A, etc) ?

I found lots of ESP32 development board on aliexpress with UNO D1 R32 form factor but they do not seems to integrate a 5V level shifter for 5V logic output to the led strip : https://fr.aliexpress.com/item/1005005046783922.html
Aside from the 5V output issue, I am not even sure those UNO form factor ESP32 boards are really supported by HyperSerial32.

**Comment by olegeras** [comment]
you complicate things, the developer came up with everything for us, just buy s2mini 2$ + level converter 3.3v>5v 0.5$ solder 4 wires and everything works great

**Reply by computergreek** [reply]
heads up, some of us (myself definitely included) are really new at this. i've never touched an ESP board in my life and have the same question as OP
